### PR TITLE
Dev ntu pvlib

### PR DESCRIPTION
**Pull request by csl37**
Ming Chuan's version of the PVLibAgent for the NTU use-case with some further modifications

### CODE REVIEW

File: Agents/NTUPVLibAgent/README.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# NTUPVLib Agent
+
+The NTUPVLib Agent is a modified version of the PVLibAgent developed to work with the PV data available for the NTU use case. The agent assumes a standard PV for the purpose of running PVLib and then scales the output by the PV area for the building. An additional scale factor is included to scale the data to a magnitude appropriate for the 15-bus NTU test system.
+
+This agent is designed to calculate AC and DC Power output from Photovaltaic Panels based on values provided in the properties files or values queried from the knowledge graph. It will then initialise the AC and DC Power as timeseries in the knowledge graph. The agent uses the [time-series client](https://github.com/cambridge-cares/TheWorldAvatar/tree/develop/JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries) from the JPS_BASE_LIB to interact with both the knowledge graph and database and uses [PvLib](https://pvlib-python.readthedocs.io/en/stable/) for it's AC and DC Power calculations.
+
+For the agent to read data, three property files are required:
```
Comment (sganguly-cares) [inline Agents/NTUPVLibAgent/README.md:7]: four property files

File: Agents/NTUPVLibAgent/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+- `strings_per_inverter` the number of strings per inverter
+
+##### Misc properties
+
+The agent takes a scale factor that can be applied to the calculated power. The scale_factor can be set in the file `misc_parameters.properties` found in the `resources` folder.
+
+####  [Option 2] Read Photovoltaic Model Specs from Knowledge Graph
+Alternatively, the parameters required to create a solar PV Model can be read from a knowledge graph. This requires an instantiation agent to create a Knowledge Graph filled with PV model parameter values. The  [HistoricalNTUEnergy Agent](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Agents/HistoricalNTUEnergyAgent) provides an example to instantiate a knowledge graph which includes PV model parameters.
+
+This option is currently only available for latitude and longitude.
```
Comment (sganguly-cares) [inline Agents/NTUPVLibAgent/README.md:66]: I don't understand this statement - is it that this option can be used with lat/lon as input? what are other possible inputs?

File: Agents/NTUPVLibAgent/README.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
+# NTUPVLib Agent
+
+The NTUPVLib Agent is a modified version of the PVLibAgent developed to work with the PV data available for the NTU use case. The agent assumes a standard PV for the purpose of running PVLib and then scales the output by the PV area for the building. An additional scale factor is included to scale the data to a magnitude appropriate for the 15-bus NTU test system.
```
Comment (sganguly-cares) [inline Agents/NTUPVLibAgent/README.md:3]: Could you include the link to the PVLib agent?

File: Agents/NTUPVLibAgent/README.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+More information about adding custom containers to the stack can be found [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager#adding-custom-containers).
+
+##### Spin Up Stack
+Follow the [steps](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager#spinning-up-a-stack) to spin up the stack.
+
+##### Run the agent
+
+Send a GET request, with the parameters including the start and end dates and times in ISO format (UTC):
+- stack=true
+- device=openmeteo
```
Comment (sganguly-cares) [inline Agents/NTUPVLibAgent/README.md:165]: will device always be openmeteo or are there other options?

**Review by sganguly-cares — CHANGES_REQUESTED** [review]
Changes needed for the readme

File: Agents/NTUPVLibAgent/README.md (hunk 4 above)
Comment (csl37) [reply Agents/NTUPVLibAgent/README.md:167]: always openmeteo for the NTU case

**Review by sganguly-cares — APPROVED** [review]

**Pull request merged**